adapters/socketio_win32.c

```diff
@@ -331,9 +339,14 @@ CONCRETE_IO_HANDLE socketio_create(void* io_create_parameters)
                 else
                 {
                     result->addrInfo = calloc(1, sizeof(struct addrinfo));
+
                     if (result->addrInfo == NULL)
                     {
                         LogError("Failure: addrInfo == NULL.");
+                        if (result->hostname != NULL)
+                        {
+                            free(result->hostname);
```

Comment:
>free(result->hostname); [](start = 28, length = 23)

For here and in Berkley equivalent, could we please have a common cleanup path instead of having the free's in multiple places?  More likely to be issues here since we don't have UT error mocking on c-util stuff.

Reply:
I thought about that, but will do it separately as it would add more change to a sensitive component.

Reply:
Ok, I take it back. I'll do it.